src/views/CardStack.js

```diff
@@ -119,166 +119,60 @@ class CardStack extends Component<DefaultProps, Props, void> {
    */
   _immediateIndex: ?number = null;
 
-  static Card = Card;
-  static Header = Header;
-
-  static propTypes = {
-    /**
-     * Custom style applied to the card.
-     */
-    cardStyle: PropTypes.any,
-
-    /**
-     * Style of the stack header. `float` means the header persists and is shared
-     * for all screens. When set to `screen`, each header is rendered within the
-     * card, and will animate in together.
-     *
-     * The default for `modal` mode is `screen`, and the default for `card` mode
-     * is `screen` on Android and `float` on iOS.
-     */
-    headerMode: PropTypes.oneOf(['float', 'screen', 'none']),
-
-    /**
-     * Custom React component to be used as a header
-     */
-    headerComponent: PropTypes.func,
-
-    /**
-     * Style of the cards movement. Value could be `card` or `modal`.
-     * Default value is `card`.
-     */
-    mode: PropTypes.oneOf(['card', 'modal']),
-
-    /**
-     * The distance from the edge of the card which gesture response can start
-     * for. Default value is `30`.
-     */
-    gestureResponseDistance: PropTypes.number,
-
-    /**
-     * Optional custom animation when transitioning between screens.
-     */
-    transitionConfig: PropTypes.func,
-
-    /**
-     * The navigation prop, including the state and the dispatcher for the back
-     * action. The dispatcher must handle the back action
-     * ({ type: NavigationActions.BACK }), and the navigation state has this shape:
-     *
-     * ```js
-     * const navigationState = {
-     *   index: 0, // the index of the selected route.
-     *   routes: [ // A list of routes.
-     *     {key: 'page 1'}, // The 1st route.
-     *     {key: 'page 2'}, // The second route.
-     *   ],
-     * };
-     * ```
-     */
-    navigation: PropTypes.shape({
-      state: NavigationPropTypes.navigationState.isRequired,
-      dispatch: PropTypes.func.isRequired,
-    }).isRequired,
-
-    /**
-     * Custom style applied to the cards stack.
-     */
-    style: View.propTypes.style,
-  };
+  _screenDetails: {
+    [key: string]: ?NavigationScreenDetails,
+  } = {};
 
-  static defaultProps: DefaultProps = {
-    mode: 'card',
-    headerComponent: Header,
-  };
+  props: Props;
 
-  componentWillMount() {
-    this._render = this._render.bind(this);
-    this._renderScene = this._renderScene.bind(this);
+  constructor(props: Props) {
+    console.log('Controlled Card stack init ',props.navigation.state);
+    super(props);
   }
 
-  render() {
-    return (
-      <Transitioner
-        configureTransition={this._configureTransition}
-        navigation={this.props.navigation}
-        render={this._render}
-        style={this.props.style}
-        onTransitionStart={this.props.onTransitionStart}
-        onTransitionEnd={this.props.onTransitionEnd}
-      />
-    );
+  componentWillReceiveProps(props: Props) {
+    console.log('Controlled Card stack recieving props ',props.navigation.state);
```

Comment:
@evv we most likely want to get rid of `console.log`. There's already discussion that there's should be a way to configure it.